src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IUsingStatement.vb

```diff
@@ -192,33 +262,672 @@ Module Program
+            Dim source = <![CDATA[
+Option Strict On
+Imports System
+
+Module Program
+    Sub Main(args As String())
+        Using c1 = New C, c2 = New C'BIND:"Using c1 = New C, c2 = New C"
+        End Using
+    End Sub
+
+    Class C
+        Implements IDisposable
+
+        Public Sub Dispose() Implements IDisposable.Dispose
+        End Sub
+    End Class
+End Module]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IVariableDeclarationStatement (2 declarations) (OperationKind.VariableDeclarationStatement) (Syntax: 'Using c1 =  ...  c2 = New C')
+  IVariableDeclaration (1 variables) (OperationKind.VariableDeclaration) (Syntax: 'c1')
+    Variables: Local_1: c1 As Program.C
+    Initializer: 
+      IObjectCreationExpression (Constructor: Sub Program.C..ctor()) (OperationKind.ObjectCreationExpression, Type: Program.C) (Syntax: 'New C')
+        Arguments(0)
+        Initializer: 
+          null
+  IVariableDeclaration (1 variables) (OperationKind.VariableDeclaration) (Syntax: 'c2')
+    Variables: Local_1: c2 As Program.C
+    Initializer: 
+      IObjectCreationExpression (Constructor: Sub Program.C..ctor()) (OperationKind.ObjectCreationExpression, Type: Program.C) (Syntax: 'New C')
+        Arguments(0)
+        Initializer: 
+          null
+]]>.Value
+
+            Dim expectedDiagnostics = String.Empty
+
+            VerifyOperationTreeAndDiagnosticsForTest(Of UsingStatementSyntax)(source, expectedOperationTree, expectedDiagnostics)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact()>
+        Public Sub IUsingStatement_UsingBlockSyntax_UsingStatementSyntax_WithExpressionResource()
+            Dim source = <compilation>
+                             <file Name="a.vb">
+                                 <![CDATA[
+Option Strict On
+Imports System
+
+Module Program
+    Sub Main(args As String())
+        Dim c1 As New C
+        Using c1'BIND:"Using c1"
+            Console.WriteLine()
+        End Using
+    End Sub
+
+    Class C
+        Implements IDisposable
+
+        Public Sub Dispose() Implements IDisposable.Dispose
+        End Sub
+    End Class
+End Module]]>
+                             </file>
+                         </compilation>
+
+            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(source, parseOptions:=TestOptions.RegularWithIOperationFeature)
+            CompilationUtils.AssertNoDiagnostics(comp)
+
+            Dim tree = comp.SyntaxTrees.Single()
+            Dim node = tree.GetRoot().DescendantNodes().OfType(Of UsingBlockSyntax).Single().UsingStatement
+
+            Assert.Null(comp.GetSemanticModel(node.SyntaxTree).GetOperation(node))
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact(Skip:="https://github.com/dotnet/roslyn/issues/22362")>
+        Public Sub IUsingStatement_UsingStatementSyntax_VariablesSyntax()
```

Comment:
>IUsingStatement_UsingStatementSyntax_VariablesSyntax [](start = 19, length = 52)

Do we have similar test for C#?

Reply:
IUsingStatement_UsingStatementSyntax_Declaration is similar, though the same concept doesn't quite apply to C#, as you can't have multiple resources in a C# using.

---
In reply to: [144357011](https://github.com/dotnet/roslyn/pull/22630#discussion_r144357011) [](ancestors = 144357011)

Reply:
> you can't have multiple resources in a C# using

What about this quote from the language spec?
When a resource-acquisition takes the form of a local-variable-declaration, it is possible to acquire multiple resources of a given type. A using statement of the form
using (ResourceType r1 = e1, r2 = e2, ..., rN = eN) statement
is precisely equivalent to a sequence of nested using statements:
using (ResourceType r1 = e1)
	using (ResourceType r2 = e2)
		...
			using (ResourceType rN = eN)
				statement